### ISSUE TITLE
rust: fails on projects with crates located in sub directories of length > 1 

### ISSUE THREAD

**Issue by DavHau** [nix-community/dream2nix#81]
flake.nix to reproduce the issue:
```nix
{
  inputs.dream2nix.url = "github:nix-community/dream2nix";
  inputs.ripgrep = { flake = false; url = "github:BurntSushi/ripgrep/13.0.0"; };
  outputs = { self, nixpkgs, dream2nix, ripgrep, }@inputs:
    let
      # change according to your system
      system = "x86_64-linux";
      dream2nix = inputs.dream2nix.lib.init {
        pkgs = nixpkgs.legacyPackages."${system}";
      };
    in {
      packages = (dream2nix.riseAndShine {
        source = builtins.path {
          path = ripgrep;
          name = "ripgrep";
        };
      }).packages;
    };
}
```

Current error:
```
last 10 log lines:
       > '/nix/store/k43y0s16wzm4h477mjkcm7p9mfkjjsab-unpack-clap-2.33.3/src/fmt.rs' -> '/nix/store/60vrxbl316809dxwmrazcqk8d6kdkzc4-vendor-ripgrep-13.0.0/clap-2.33.3/src/fmt.rs'
       > '/nix/store/k43y0s16wzm4h477mjkcm7p9mfkjjsab-unpack-clap-2.33.3/src/lib.rs' -> '/nix/store/60vrxbl316809dxwmrazcqk8d6kdkzc4-vendor-ripgrep-13.0.0/clap-2.33.3/src/lib.rs'
       > '/nix/store/k43y0s16wzm4h477mjkcm7p9mfkjjsab-unpack-clap-2.33.3/src/macros.rs' -> '/nix/store/60vrxbl316809dxwmrazcqk8d6kdkzc4-vendor-ripgrep-13.0.0/clap-2.33.3/src/macros.rs'
       > '/nix/store/k43y0s16wzm4h477mjkcm7p9mfkjjsab-unpack-clap-2.33.3/src/map.rs' -> '/nix/store/60vrxbl316809dxwmrazcqk8d6kdkzc4-vendor-ripgrep-13.0.0/clap-2.33.3/src/map.rs'
       > '/nix/store/k43y0s16wzm4h477mjkcm7p9mfkjjsab-unpack-clap-2.33.3/src/osstringext.rs' -> '/nix/store/60vrxbl316809dxwmrazcqk8d6kdkzc4-vendor-ripgrep-13.0.0/clap-2.33.3/src/osstringext.rs'
       > '/nix/store/k43y0s16wzm4h477mjkcm7p9mfkjjsab-unpack-clap-2.33.3/src/strext.rs' -> '/nix/store/60vrxbl316809dxwmrazcqk8d6kdkzc4-vendor-ripgrep-13.0.0/clap-2.33.3/src/strext.rs'
       > '/nix/store/k43y0s16wzm4h477mjkcm7p9mfkjjsab-unpack-clap-2.33.3/src/suggestions.rs' -> '/nix/store/60vrxbl316809dxwmrazcqk8d6kdkzc4-vendor-ripgrep-13.0.0/clap-2.33.3/src/suggestions.rs'
       > '/nix/store/k43y0s16wzm4h477mjkcm7p9mfkjjsab-unpack-clap-2.33.3/src/usage_parser.rs' -> '/nix/store/60vrxbl316809dxwmrazcqk8d6kdkzc4-vendor-ripgrep-13.0.0/clap-2.33.3/src/usage_parser.rs'
       > '/nix/store/k43y0s16wzm4h477mjkcm7p9mfkjjsab-unpack-clap-2.33.3/.cargo-checksum.json' -> '/nix/store/60vrxbl316809dxwmrazcqk8d6kdkzc4-vendor-ripgrep-13.0.0/clap-2.33.3/.cargo-checksum.json'
       > cp: cannot stat '/nix/store/plm64awd1mip3pj8lm9v030ywpdamzx9-ripgrep/grep': No such file or directory
```

Even after fixing the issue in the translator (done in #80 ), it still fails with another error about not finding some source.

@yusdacra

**Comment by yusdacra** [comment]
I got the same error with the flake here. I changed the dream2nix input to use the PR branch but then i started getting `error: flake output attribute 'packages.x86_64-linux.ripgrep' is not a derivation`.

**Comment by DavHau** [comment]
The other branch is merged now and the problem with the output fixed via 2167b7e993abae2abb4d8459888b46a1cf1c9384
